src/android/com/adobe/phonegap/push/GCMIntentService.java

```diff
@@ -58,8 +60,19 @@ public void setNotification(int notId, String message){
     }
 
     @Override
-    public void onMessageReceived(String from, Bundle extras) {
-        Log.d(LOG_TAG, "onMessage - from: " + from);
+    public void onMessageReceived(RemoteMessage message){
+
+        Log.d(LOG_TAG, "onMessage - from: " + message.getFrom());
+
+        Bundle extras = new Bundle();
+
+        if (message.getNotification()!=null) {
+            extras.putString(TITLE,message.getNotification().getTitle());
+            extras.putString(MESSAGE,message.getNotification().getBody());
+        }
+        for (Map.Entry<String, String> entry : message.getData().entrySet()) {
+            extras.putString(entry.getKey(), entry.getValue());
+        }
```

Comment:
Hmmm....need to look into RemoteMessage some more but looks like a good idea to minimize changes for now.